### PR TITLE
InfluxDB 2.x support

### LINKED ISSUE #255: influxunifi: Add support for InfluxDB 2.0 output
While still currently in beta, i've been using [InfluxDB 2.0](https://v2.docs.influxdata.com/v2.0/) for a couple of months now and rather enjoy it. Grafana has recently [added native InfluxQL & Flux support with v7.1](https://grafana.com/blog/2020/07/17/grafana-v7.1-released-new-features-for-influxdb-and-elasticsearch-data-sources-table-panel-transformations-and-more/#influx-data-source), and it looks like feature development for InfluxDB 1.x has stopped.

### LINKED ISSUE #407: Add Influxdb v2 workaround to docs
Hi

UniFi Poller can talk to InfluxDB 2.0 using DBRP mappings : https://docs.influxdata.com/influxdb/v2.0/tools/grafana/?t=InfluxQL#view-and-create-influxdb-dbrp-mappings

Once your InfluxDB 2.0 bucket is created (using CLI or 8086 web UI) :

```
# influx bucket list
ID                      Name                    Retention       Organization ID
f2763f172b50dea3        _monitoring             168h0m0s        143f24876e1ec5d2
74e9f584fccacedc        _tasks                  72h0m0s         143f24876e1ec5d2
7112dfae5d1aa164        unifi                   8760h0m0s       143f24876e1ec5d2

# influx v1 dbrp create --db unifi-db --rp unifi-rp --bucket-id  7112dfae5d1aa164 --default
ID                      Database        Bucket ID               Retention Policy        Default Organization ID
075053f3abe50000        unifi-db        7112dfae5d1aa164        unifi-rp                true    143f24876e1ec5d2

# influx v1 auth create  --read-bucket 7112dfae5d1aa164 --write-bucket 7112dfae5d1aa164 --username unifi-user
Please type your password: 

Please type your password again:
```

You then just have to fill up.conf :

```
[influxdb]
  disable = false
  url  = "http://influxdb:8086"
  user = "unifi-user"
  pass = "mysuperpassword"
  db = "unifi-db"
```

_Originally posted by @Fredouye in https://github.com/unpoller/unpoller/issues/255#issuecomment-812365646_

### LINKED ISSUE #430: Upgraded to influx v2 and some data is now missing.
I updated my influxdb to v2 and noticed that now some graphs are empty (UAP Bandwith, Received signal strength, wireless signal  and many others).  No errors reported on the service:

    [INFO] UniFi Metrics Recorded. Site: 1, Client: 58, UAP: 2, UDM/USG/: 0, USW: 3, IDS/Event/Alarm/Anomaly: 0/0/0/0, DPI Site/Client: 0/0, Point: 110, Fields: 3360, Err: 0, Dur: 13ms

I do have some data such as switch temperature, fan level, memory, data transfer and others. 

Any ideas?

### PR DESCRIPTION
**Pull request by platinummonkey**
- fixes #430 
- fixes #407
- fixes #255 

the config is slightly changed

### CODE REVIEW

File: pkg/influxunifi/influxdb.go [hunk 1]
```diff
@@ -22,22 +23,35 @@ import (
...
-	URL       string        `json:"url,omitempty" toml:"url,omitempty" xml:"url" yaml:"url"`
-	User      string        `json:"user,omitempty" toml:"user,omitempty" xml:"user" yaml:"user"`
-	Pass      string        `json:"pass,omitempty" toml:"pass,omitempty" xml:"pass" yaml:"pass"`
-	DB        string        `json:"db,omitempty" toml:"db,omitempty" xml:"db" yaml:"db"`
-	Disable   bool          `json:"disable" toml:"disable" xml:"disable,attr" yaml:"disable"`
-	VerifySSL bool          `json:"verify_ssl" toml:"verify_ssl" xml:"verify_ssl" yaml:"verify_ssl"`
+	Interval cnfg.Duration `json:"interval,omitempty" toml:"interval,omitempty" xml:"interval" yaml:"interval"`
+	Version2 bool          `json:"version2,omitempty" toml:"version2,omitempty" xml:"version2" yaml:"version2"`
+
+	// V1
```
Comment (davidnewhall) [inline pkg/influxunifi/influxdb.go:40]: The way godoc works, this sets the doc for `User` to `V1`.  I do not recommend a single comment like this for multiple struct members. Add a comment-per-struct member to produce beautiful code docs. 

Also, V2 doesn't use user/pass/db?

File: pkg/influxunifi/influxdb.go [hunk 2]
```diff
@@ -22,22 +23,35 @@ import (
...
 type Config struct {
-	Interval  cnfg.Duration `json:"interval,omitempty" toml:"interval,omitempty" xml:"interval" yaml:"interval"`
-	URL       string        `json:"url,omitempty" toml:"url,omitempty" xml:"url" yaml:"url"`
-	User      string        `json:"user,omitempty" toml:"user,omitempty" xml:"user" yaml:"user"`
-	Pass      string        `json:"pass,omitempty" toml:"pass,omitempty" xml:"pass" yaml:"pass"`
-	DB        string        `json:"db,omitempty" toml:"db,omitempty" xml:"db" yaml:"db"`
-	Disable   bool          `json:"disable" toml:"disable" xml:"disable,attr" yaml:"disable"`
-	VerifySSL bool          `json:"verify_ssl" toml:"verify_ssl" xml:"verify_ssl" yaml:"verify_ssl"`
+	Interval cnfg.Duration `json:"interval,omitempty" toml:"interval,omitempty" xml:"interval" yaml:"interval"`
+	Version2 bool          `json:"version2,omitempty" toml:"version2,omitempty" xml:"version2" yaml:"version2"`
```
Comment (davidnewhall) [inline pkg/influxunifi/influxdb.go:38]: Can you poll influx to figure this out?

**Review by davidnewhall — COMMENTED** [review]
I definitely like this more than the other PR.

File: pkg/influxunifi/influxdb.go (hunk 1 above)
Comment (platinummonkey) [reply pkg/influxunifi/influxdb.go:40]: Looks like the influx docs say auth token. Likely to be able to limit to write points only would be my guess. But I’ll admit I don’t use influx myself

File: pkg/influxunifi/influxdb.go (hunk 2 above)
Comment (davidnewhall) [reply pkg/influxunifi/influxdb.go:38]: Perhaps you can determine if V2 based on the inputs?
If username and password then v1. If auth token, then v2.

Comment (platinummonkey) [reply pkg/influxunifi/influxdb.go:38]: this might be the simplest way yes

**Pull request merged**